container/src/container/array_dict.py

```diff
@@ -9,196 +9,194 @@ def isiterable(obj):
         return False
 
 class array_dict:
-    def __init__(self, values= np.array([]), keys = None):
-        if isinstance(values,array_dict):
-            from copy import deepcopy
-            self._values = deepcopy(values._values)
-            self._keys = deepcopy(values._keys)
-            self._index = deepcopy(values._index)
-        elif isinstance(values,dict):
-            self._values = np.array(values.values())
-            self._keys = np.array(values.keys())
-            #self._index = np.array(nd.sum(np.arange(len(values)),self._keys,index=np.arange(max(self._keys)+1)),int)
-            self._index = np.zeros(self._keys.max()+1,int)
-            self._index[self._keys] = np.arange(len(self._keys))
-        elif isinstance(values,list) and isinstance(values[0],tuple):
-            self._values = np.array(values)[:,1]
-            self._keys = np.array(np.array(values)[:,0],int)
-            #self._index = np.array(nd.sum(np.arange(len(values)),self._keys,index=np.arange(max(self._keys)+1)),int)
-            self._index = np.zeros(self._keys.max()+1,int)
-            self._index[self._keys] = np.arange(len(self._keys))
-        else:
-            self._values = np.array(values)
-            if not keys is None:
-                assert len(keys) == len(values)
-                self._keys = np.array(keys)
-                #self._index = np.array(nd.sum(np.arange(len(values)),self._keys,index=np.arange(max(self._keys)+1)),int)
-                self._index = np.zeros(self._keys.max()+1,int)
-                self._index[self._keys] = np.arange(len(self._keys))
-            else:
-                if len(values)>0:
-                    self._keys = np.arange(len(values))
-                    self._index = np.arange(len(values))
-                else:
-                    self._keys = np.array([],int)
-                    self._index = np.array([],int)
-
-    def __getitem__(self,key):
-        if self._index[key] > 0:
-            return self._values[self._index[key]]
-        elif key in self._keys:
-            return self._values[0]
-        else:
-            raise KeyError(str(key))
-
-    def __setitem__(self,key, value):
-        import numpy as np
```

Comment:
Try to avoid internal import.
**PEP 08 states**:
    Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.